### DISCUSSION TITLE
Help Flashing/Supporting New SNES Repro - M29W640FT? (Chinese repro board)

### DISCUSSION THREAD

**Discussion by trgkaigit** [sanni/cartreader#276] [Need help]
I received a Chinese SNES repro board the other day that doesn't use the more common L-02-rst-10k or H01-10k boards I've used the cart reader to reflash, and unfortunately this one isn't recognized.  I'm curious if this is something that can be supported in a future release.  I've attached a picture of the board.  I believe the chip that matters for the purposes of support is the M29W640FT?
![image1](https://user-images.githubusercontent.com/92490546/137226137-a6f4fbac-857c-418d-9ba2-9f80124b5eef.jpeg)
![image0](https://user-images.githubusercontent.com/92490546/137226144-e338d40f-9eb5-4eee-9f34-92a2e4615a80.jpeg)

**Comment by StanMOI** [comment]
Its should be ExHIROM board .
Do you tried to see if pin 11 of flash chip are connected to pin 54 of cartridge edge?

**Reply by trgkaigit** [reply]
When I select ExHiROM from the Flash Repro menu, I get:

ID Type 1: 0000
ID Type 2: 0000
UNKNOWN FLASHROM

My other boards so far let me flash without any extra steps.  I don't have my multimeter in yet to test if Pin 11 on the flash connects to Pin 54 on the cart.  Will I have to wire the two pins together to enable rewriting this board?

**Reply by StanMOI** [reply]
Try first to find connections between : WE (pin 11 of Flash) and WR (pin 54 of cartridge edge).
Maybe it can resolve your problem.
To be able to flash your chip they have to be connected WE to WR.
After flashing WE (pin 11 of flash chip) have to be reconnected to VCC.
I'am not sure but I dont think that is it your problem.
Anyway better to be sure.

I have the same problem with 9090 error code (UNKNOWN FLASHROM ), with good connection to be flashed and supported flash chip on my board.
Chines sellers flash those PCB with 6 holes (16-byte mode flashing).
Sanni cart reader do flashing via the cartridge EDGE of pcb.
So i'am not sure, but cause of some resistors or capacitors some of address pins can't be detected.
I think my problem are the same but with different error code.
Or maybe you flash chip are not supported at all.
Sanni can answer you better than me.

Can you give me link to seller for this PCB, please.
I will order and see.

**Reply by trgkaigit** [reply]
Unfortunately the seller doesn't sell PCBs directly and they don't seem to be consistent on which boards are used.  Some copies came on the board pictured above, while the next batch came on the H10-10K boards which are fully supported by the Sanni Cartreader.  I found them here: https://www.alibaba.com/product-detail/16-Bit-English-Language-Battery-Save_1600095133766.html

I'd love to just buy boards directly, but all I've found so far are blank PCBs which require buying the chips and soldering each one together, while this was a way to get a bunch of easily reprogrammable boards at ~$7/each after shipping, and it includes a shell that requires screws, which would normally cost me an extra $3-4 on top (or require sacrificing a real cart).

**Reply by StanMOI** [reply]
Thanks for link.
Order is 5 minimum :(
Just to do some research, it's to expansif for me.

By the way...
Do you know that those Chines PCB are out of specification for SNES system ?
They are very chip, yes, but it will slowly damage your SNES system and some of them can damage cartridge also...
So, to make business it's maybe a good way, but if you love your SNES and are looking for SNES cartridges that will be save for using, you wouldn't find it in China.

**Comment by Gandalf01980** [comment]
Hello

Just got a same PCB with M29W640FT, same problem with the latest v15 version. Anyone find the way to reflash this PCB with sanni?

Thanks

**Reply by StanMOI** [reply]
As i remember, M29W640FT need 12v to flashing. So it’s not supported by Sanni programmer .

**Reply by Chomemel** [reply]
Nope. It only needs around 3V to be programmed (maybe less, I don't remeber the datasheet). You can flash it with the CFI command.

But here is the weird thing. I had a chinese repro with a M29W640 and the OSCR was unable to identify/read/flash it.
I unsoldered the M29W640 and replaces it with a 29LV640. Now the repro works like a charm.
And for the M29W640, I fashes it with the OSCR using the CFI command. Then solder it to a real SNES cartirdge (via an adapter). And it works just fine.

I never really undertood...

**Reply by Gandalf01980** [reply]
thanks for the info!

**Comment by StanMOI** [comment]
There is no code line for Flash type M29W640 inside of [FLASH.ino](https://github.com/sanni/cartreader/blob/master/Cart_Reader/FLASH.ino)
Code line need to be added manually or ask Sanni to add flash type M29W.

**Reply by Chomemel** [reply]
I tried ast year, to add lines to the code but I couldn't succeed as the chip would never be identified by the OSCR. But I'm far to be an expert.
